src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -2390,7 +2390,7 @@ class CgroupUtils:
                 pbs.logmsg(pbs.EVENT_DEBUG4, "Insufficient ngpus: %s/%s" %
                            (ngpus, gpus))
                 return {}
-            names, devices = self.__assign_devices('gpu', gpus[:tmp_ngpus],
+            names, devices = self.__assign_devices('gpu', gpus[:ngpus],
```

Comment:
Mike I suggest that you run the script through pylint, I ran it and it reported a bunch of problems like on line 1621  system_info is undefined, on line 3009 time is undefined, 3351 'job' is undefined, on line 3644 we except JSONDecodeError but this error is only present in python 3.5 and many more.